staging/src/k8s.io/apiextensions-apiserver/pkg/features/kube_features.go

```diff
@@ -47,6 +49,34 @@ func init() {
 	utilfeature.DefaultMutableFeatureGate.Add(defaultKubernetesFeatureGates)
 }
 
+var (
+	CRDValidationRatcheting2 = featuregates.NewFeatureGate("CRDValidationRatcheting").
+					EnableByDefault().
+					Beta().
+					ToFeatureGateOrDie()
+
+	CustomResourceFieldSelectors2 = featuregates.NewFeatureGate("CustomResourceFieldSelectors").
+					Beta().
+					ToFeatureGateOrDie()
+
+	libraryFeatureSet = featuregates.NewSimpleFeatureSet()
+)
+
+func init() {
+	libraryFeatureSet.AddFeatureGatesOrDie(
+		CRDValidationRatcheting2,
+		CustomResourceFieldSelectors2,
+	)
+	libraryFeatureSet.AddFeatureSetsOrDie(
+		clientfeatures.LibraryFeatureSet(),
+	)
+}
+
+// notice that this narrows the interface from above
+func LibraryFeatureSet() featuregates.FeatureSet {
```

Comment:
we want a narrower interface for composition.  Turns out, we no longer have a reason to expose the full featureset other than composition.  Perhaps we want to expose some interface for code level inspection of gates, but it's not "naturally" necessary.

Reply:
I don't follow this comment?

Reply:
> I don't follow this comment?

This method returns an interface with a much smaller surface area than the entire implementation.  Just in contrast to where are now.